test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNetCore3x/BulkOperation/EFTests/BulkOperationTestEF.cs

```diff
@@ -0,0 +1,285 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="BulkOperationTestEF.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNet.OData;
+using Microsoft.AspNet.OData.Extensions;
+using Microsoft.AspNet.OData.Routing;
+using Microsoft.AspNet.OData.Routing.Conventions;
+using Microsoft.Test.E2E.AspNet.OData.Common.Execution;
+using Microsoft.Test.E2E.AspNet.OData.Common.Extensions;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Microsoft.Test.E2E.AspNet.OData.BulkOperation
+{
+    public class BulkOperationTestEF : WebHostTestBase
+    {
```

Comment:
can you add tests for when the failed scenarios will be called. Like when an insert was not successful, how is the failed scenario handled and the expected response. Add for all the scenarios.

Reply:
Not adding EF tests now